tests/layer_tests/ovc_python_api_tests/test_tf.py

```diff
@@ -1287,3 +1288,24 @@ def test_tf1_input_with_identity(self):
 
         assert ov_model.inputs[0].get_names() == {"x:0"}
         assert ov_model.inputs[1].get_names() == {"y:0"}
+
+
+class TestUnicodePaths(unittest.TestCase):
+    @pytest.mark.nightly
+    @pytest.mark.precommit
+    def test_unicode_paths(self):
+        test_directory = os.path.dirname(os.path.realpath(__file__))
+        with tempfile.TemporaryDirectory(dir=test_directory, prefix=r"晚安_путь_к_файлу") as temp_dir:
+            model, model_ref, _ = create_tf_graph_def(None)
+            model_path = save_to_pb(model, temp_dir)
+
+            from openvino import convert_model
+            res_model = convert_model(model_path)
+            flag, msg = compare_functions(res_model, model_ref, False)
+            assert flag, msg
+
+            from openvino.frontend import FrontEndManager
+            fm = FrontEndManager()
+            fe = fm.load_by_framework("tf")
```

Comment:
Will it work if we do the same in C++ code?

Reply:
It works, but only if we pass correctly created `wstring`. 

It looks like expected behavior, as `string` class works correctly only with single-byte characters, while unicode symbols are larger https://cplusplus.com/reference/string/string/

But if we pass wstring, all works https://cplusplus.com/reference/string/wstring/